src/hooks/useTodos.ts

```diff
@@ -0,0 +1,57 @@
+import { useState, useEffect, SetStateAction, Dispatch, useRef } from 'react';
+import { Todo } from '../types/Todo';
+import { getTodos } from '../api/todos';
+
+export type UseTodos = {
+  todos: Todo[];
+  setTodos: Dispatch<SetStateAction<Todo[]>>;
+  errorMessage: string;
+  setErrorMessage: (error: string) => void;
+  tempTodo: Todo | null;
+  setTempTodo: (todo: Todo | null) => void;
+  loadingIds: number[];
+  setLoadingIds: Dispatch<SetStateAction<number[]>>;
+};
```

Comment:
This looks like an overhead

You can simply remove this type and still benefit of all Typescrit things

Reply:
> setTodos: (value: Todo[]) => void; - in this case i have problems when func setstate

I missed, that you are using both versions of the setter
then it is ok to keep your version